docs/adr/0043-react-functional-tests.md

```diff
@@ -0,0 +1,170 @@
+- signin is a pre-requisite for many functional tests, and this setup is currently configured to use the backbone flow
+- use of distinct locators for backbone and react, with backbone locators not working for react and vice-versa
+- differences in flow between backbone and react
+
+Going forward, what cohesive approach should we adopt in order to ease rollout of React flows while maximizing coverage and minimizing impact on timelines?
+
+### Previous react rollouts
+
+- ResetPassword: Fully rolled out in production, backbone routes no longer used. All backbone tests are skipped. Only the new `react-conversion` tests are running. These tests were relatively well isolated and the migration strategy worked well.
+- Signup: Full prod rollout was previously attempted and rolled back, second full prod rollout is planned for Train 283 (stage on 2024/04/24, production on 2024/05/01). New react tests were created, and where the signup flow was used in non-signup tests a condition was added with test steps to go through either flow based on feature flag. In preparation for the initial full prod rollout, backbone signup tests were skipped when the feature flag was on. By consequence, with the feature flag currently on for all environments, all backbone tests are skipped even though 85% of users see the backbone version of the signup flow.
+
+### Migration strategy and challenges
+
+Up until now, the primary strategy for supporting react flows with functional tests has mainly been to:
+
+- create new react-specific POMs (page object models)
+- duplicate tests and place them in a `react-conversion` directory (tests selected for duplication are based on the [Playwright Migration Analysis][2])
+- conditionally run these react tests based on a feature flag indicating the experiment is enabled for the route group
+
+Additional strategies that have been used include:
+
+- using conditional logic within tests to switch between react and backbone and react for a portion of the tests steps, based on feature flag configuration
+- using conditional logic within tests to run through both backbone and react versions of flows.
+- skip the backbone tests for full prod rollout (but there is no configuration value that can be checked)
+
+Additional challenges:
+
+- competing ideas have been implemented to bridge the gaps between React and Backbone in the functional tests
+- the desired end state of the react functional test migration is not universally agreed upon
+- Multiple react routes at different phases of rollout means that conditional logic is required to navigate between react/backbone test steps
+- The tests in `react-conversion` will need to be modified again during the cleanup phase to remove all of the feature flag check and experiment parameters
+
+### Questions
+
+- Are there new pattern that would simplify conversion moving forward (e.g., user-facing locators, single page POMs, route check in POMs, etc.)
```

Comment:
Not sure, this might just be a consequence of how our feature flagging worked and our design decisions.

Reply:
That's true, the reason that some locators that were set for Backbone no longer work is because of design decisions in React. However using user-facing locators (instead of test ids or css selectors) can in many cases (but not all) be compatible for both Backbone and React. 

This is probably an overall test architecture question though.